src/containers/App/App.js

```diff
@@ -85,42 +88,55 @@ const App = (props) => {
   if (location.pathname === `${urlLink}/ledgers`) {
     return <Redirect to={urlLink} />
   }
+  const queryClient = new QueryClient({
+    defaultOptions: {
+      queries: {
+        refetchOnWindowFocus: false,
+        refetchOnReconnect: false,
+        refetchOnMount: false,
+        retry: false,
+      },
+    },
+  })
```

Comment:
what is the purpose of react query here? what does it add over just calling the API directly ?

Reply:
This just sets up react-query for the app. @shadymoses might be able to expound on its benefits better than I.

Reply:
From their website:

> gives you declarative, always-up-to-date auto-managed queries and mutations that directly improve both your developer and user experiences

Makes your requests reusable hooks with a shared data cache. No more manual fetching or re-fetching, or messing with something like redux to manage server state. The explorer's current redux store is made up of 2 things:

1. some app measuring data (which isn't really being used and could be done much simpler)
2. request data that is mostly not shared across pages

All of the (necessary) app state is server state. This is not a good use case for redux. `react-query` makes handling server state significantly nicer in every regard.

Reply:
yeah i saw their site and i _completely_ agree that we have no need for redux in our application. my question is why we _do_ need react query, and i'm still not seeing it. i don't see why we have a need for anything special in the app storage world

Reply:
i think it is having to avoid the need of creating hooks to manage states like loading, error, etc

Reply:
Well, you have state, so it has to be managed somehow. The explorer is fairly data heavy.

It's a matter of putting in the work to roll your own:
- context providers with custom hooks, caching, managing request lifecycle states, and all the other features you might need
- or copy pasta to add manual request handling everywhere in the app that needs data

Or you could use an amazingly well crafted existing solution. Savings in time and quality all around.